src/controllers/authController.ts

```diff
@@ -0,0 +1,95 @@
+import jwt, { JwtPayload, TokenExpiredError } from 'jsonwebtoken';
+import bcrypt from 'bcryptjs';
+import { validationResult } from 'express-validator';
+import { Request, Response } from 'express';
+import { Types } from 'mongoose';
+
+import User from '../models/user.js';
+import Role from '../models/role.js';
+import { secret } from '../config/tokenKey.js';
+
+const generateAccessToken = (id: Types.ObjectId, roles: string[]) => {
+    const payload = {
+        id,
+        roles,
+    };
+    return jwt.sign(payload, secret, { expiresIn: '120000ms' });
+};
+
+class authController {
+    async registration(req: Request, res: Response) {
+        try {
+            const erorrs = validationResult(req);
+            if (!erorrs.isEmpty()) {
+                return res.status(400).json({ message: 'Registartion erorr', erorrs });
+            }
+            const { firstName, lastName, password, email } = req.body;
+            const candidate = await User.findOne({ email });
+            if (candidate) {
+                return res.status(400).json({ message: 'User already exists' });
+            }
+            const hashPassword = bcrypt.hashSync(password, 7);
+            const userRole = await Role.findOne({ value: 'USER' });
+            const user = new User({
+                firstName,
+                lastName,
+                password: hashPassword,
+                email,
+                roles: [userRole?.value],
+            });
+            await user.save();
+            const currentUser = await User.findOne({ email });
```

Comment:
after saving you can use an already existing `user`, you don't need to download it again
The `_id` will be filled for this user object, but need to check😅
In case a user with the same email already exists in the database, the `save` method throws an exception